scandium-core/src/main/java/org/eclipse/californium/scandium/dtls/cipher/CbcBlockCipher.java

```diff
@@ -194,4 +222,37 @@
 		hmac.update(content, 0, length);
 		return hmac.doFinal();
 	}
+
+	/**
+	 * Check padding.
+	 * 
+	 * The check is implemented using a "time constant" approach by always
+	 * comparing 256 bytes.
+	 * 
+	 * @param padding padding to be checked
+	 * @param data data to be checked. Must contain at least 256 + 1 bytes from the
+	 *            offset on. The value of the last byte will be changed!
+	 * @param offset offset of the padding field.
+	 * @return {@code true}, if padding bytes in data from the offset on
+	 *         contains the value of the padding byte.
+	 * @throws IllegalArgumentException if the data array doesn't contain
+	 *             257 bytes after the offset.
+	 */
+	public static boolean checkPadding(int padding, byte[] data, int offset) {
+		if (data.length < offset + 257) {
+			throw new IllegalArgumentException("data must contain 257 bytes from offset on!");
+		}
+		byte result1 = 0;
+		byte result2 = 0;
+		byte pad = (byte) padding;
+		for (int index = 0; index <= padding; ++index) {
+			result1 |= pad ^ data[offset + index];
+		}
+		for (int index = padding + 1; index < 256; ++index) {
+			result2 |= pad ^ data[offset + index];
+		}
+		// apply result2 at the "oversize" dummy data
+		data[data.length - 1] ^= result2;
```

Comment:
what is the purpose of this line ?

Reply:
Ensure, that "optimization" doesn't skip the dummy compare loop.

Reply:
(could we add this as comment ?)

Reply:
Sure.

Reply:
Done.